### PR TITLE
fix: uri and request parsing;

### PR DESCRIPTION
**Pull request by mohamed-gougam**
### Proposed changes

Right now we have a problem whenever the URI logged by NAP contains a comma `,`. That's due to the fact that we are separating logged values with comma `,`, so when splitting to get each value separately we fall in having erroneous values for both URI and request.

This fixes the problem as follow:

1. Under `/etc/app_protect/conf/log_default.json` add config to escape the comma `,` for each logged value with the URI encoding `%2C` (this represents standard URI encoding for `,` )
2. When parsing, for both `URI` and `request`, replace `%2C` with `,`.

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [x] If applicable, I have added tests that prove my fix is effective or that my feature works
- [x] If applicable, I have checked that any relevant tests pass after adding my changes
- [x] I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))

### CODE REVIEW

File: src/plugins/nap_monitoring.go [hunk 1]
```diff
@@ -55,7 +55,7 @@ func NewNAPMonitoring(env core.Environment, cfg *config.Config) (*NAPMonitoring,
 		log.Warnf("NAP Monitoring report count must be between %v and %v. Defaulting to %v",
 			minReportCountDelimiter,
 			maxReportCountDelimiter,
-			config.Defaults.NAPMonitoring.ReportInterval)
+			config.Defaults.NAPMonitoring.ReportCount)
```
Comment (sanathkumarbs) [inline src/plugins/nap_monitoring.go:58]: Good catch! Probably should add a unit test for things like this to catch and validate sooner.

**Review by sanathkumarbs — APPROVED** [review]
LGTM! We should make some Control Plane and Docs changes. I'll ping you about it.

File: src/plugins/nap_monitoring.go (hunk 1 above)
Comment (olli-holmala) [reply src/plugins/nap_monitoring.go:58]: +1 on adding unit tests

File: src/extensions/nginx-app-protect/monitoring/processor/nap.go [hunk 2]
```diff
@@ -506,13 +505,13 @@ func setValue(napConfig *NAPConfig, key, value string, logger *logrus.Entry) err
...
 	case httpURI:
-		napConfig.HTTPURI = value
+		napConfig.HTTPURI = strings.ReplaceAll(value, encodedComma, decodedComma)
 	case isTruncated:
 		napConfig.IsTruncated = value
 	case policyName:
 		napConfig.PolicyName = value
 	case request:
-		napConfig.Request = value
+		napConfig.Request = strings.ReplaceAll(value, encodedComma, decodedComma)
```
Comment (olli-holmala) [inline src/extensions/nginx-app-protect/monitoring/processor/nap.go:514]: I have a question. So are the `HTTPURI` and `Request` fields fully URL encoded or just the commas? If they are fully encoded, should we be decoding them? The implementation we have now only decodes commas.
```go
request, err := url.QueryUnescape(value)
```

**Review by olli-holmala — APPROVED** [review]

File: src/extensions/nginx-app-protect/monitoring/processor/nap.go (hunk 2 above)
Comment (mohamed-gougam) [reply src/extensions/nginx-app-protect/monitoring/processor/nap.go:514]: only the comma is encoded in `httpURI` and `request`

**Comment by oliveromahony** [comment]
waiting on some unit tests before merging this

**Review by dhurley — APPROVED** [review]

**Review by oliveromahony — APPROVED** [review]

**Pull request merged**